pkg/api/pod/util.go

```diff
@@ -575,6 +577,66 @@ func dropDisabledEphemeralVolumeSourceAlphaFields(podSpec, oldPodSpec *api.PodSp
 	}
 }
 
+func dropPodAffinityTermNamespaceSelector(terms []api.PodAffinityTerm) {
+	for i := range terms {
+		terms[i].NamespaceSelector = nil
+	}
+}
+
+func dropWeightedPodAffinityTermNamespaceSelector(terms []api.WeightedPodAffinityTerm) {
+	for i := range terms {
+		terms[i].PodAffinityTerm.NamespaceSelector = nil
+	}
+}
+
+// dropDisabledPodAffinityTermFields removes disabled fields from PodSpec related
+// to PodAffinityTerm only if it is not already used by the old spec
+func dropDisabledPodAffinityTermFields(podSpec, oldPodSpec *api.PodSpec) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.PodAffinityNamespaceSelector) &&
+		podSpec != nil && podSpec.Affinity != nil &&
+		!podAffinityNamespaceSelectorInUse(oldPodSpec) {
```

Comment:
In theory, this condition should be evaluated for each place where the API is used.
Otherwise, if the feature gets disabled but the Pod had one NamespaceSelector, then that single instance would allow to add NamespaceSelectors to other affinity terms.

However, it doesn't really matter, as affinity terms are immutable fields.

Reply:
exactly, since they are immutable, it doesn't matter.

Reply:
I would prefer this code to be made future proof. Immutability can be more easily changed than validations.

Reply:
this is the pattern used throughout the file though.

Reply:
actually, this is not validation it's strategy, so it's also possible to change in the future.

You can leave it, unless api-reviewer disagrees.

Reply:
yes, this will actually go away when the feature graduates.